src/main/java/seedu/address/logic/parser/PinCommandParser.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.PinCommand;
+import seedu.address.logic.messages.PinMessages;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Name;
+
+/**
+ * Parses input arguments and creates a new PinCommand object
+ */
+public class PinCommandParser implements Parser<PinCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the PinCommand
+     * and returns an PinCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public PinCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        Name name;
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME);
+
+        // check for missing name
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)) {
+            throw new ParseException(String.format(PinMessages.MESSAGE_PIN_MISSING_NAME,
+                    PinCommand.MESSAGE_USAGE));
+        }
+
+        name = mapName(argMultimap);
+
+        return new PinCommand(name);
+    }
+
+    /**
+     * Returns name value using PREFIX.
+     * @param argMultimap Object that contains mapping of prefix to value.
+     * @return Returns object representing name.
+     * @throws ParseException Thrown when command is in invalid format.
+     */
+    public Name mapName(ArgumentMultimap argMultimap) throws ParseException {
+        try {
+            return ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(PinMessages.MESSAGE_PIN_INVALID_NAME, pe.getMessage()));
```

Comment:
Isnt this function same level as ParserUtil.parseName(), dont quite understand why need an additional function here